timestampvm/block.go

```diff
@@ -5,52 +5,71 @@ package timestampvm
 
 import (
 	"errors"
+	"fmt"
 	"time"
 
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow/choices"
 	"github.com/ava-labs/avalanchego/snow/consensus/snowman"
-	"github.com/ava-labs/avalanchego/vms/components/core"
+	"github.com/ava-labs/avalanchego/utils/hashing"
 )
 
 var (
 	errTimestampTooEarly = errors.New("block's timestamp is earlier than its parent's timestamp")
 	errDatabaseGet       = errors.New("error while retrieving data from database")
-	errDatabaseSave      = errors.New("error while saving block to the database")
 	errTimestampTooLate  = errors.New("block's timestamp is more than 1 hour ahead of local time")
-	errBlockType         = errors.New("unexpected block type")
+	errBlockNil          = errors.New("block is nil")
 
-	_ snowman.Block = &Block{}
+	_ Block = &TimeBlock{}
 )
 
+// Block interface is a wrapper around snowman.Block
+type Block interface {
+	snowman.Block
+	Initialize(bytes []byte, status choices.Status, vm *VM)
+	Data() [dataLen]byte
+}
+
 // Block is a block on the chain.
 // Each block contains:
-// 1) A piece of data (a string)
-// 2) A timestamp
-type Block struct {
-	*core.Block `serialize:"true"`
-	Data        [dataLen]byte `serialize:"true"`
+// 1) ParentID
+// 2) Height
+// 3) Timestamp
+// 4) A piece of data (a string)
+type TimeBlock struct {
+	PrntID ids.ID        `serialize:"true" json:"parentID"`  // parent's ID
+	Hght   uint64        `serialize:"true" json:"height"`    // This block's height. The genesis block is at height 0.
+	Tmstmp int64         `serialize:"true" json:"timestamp"` // Time this block was proposed at
+	Dt     [dataLen]byte `serialize:"true" json:"data"`      // Arbitrary data
```

Comment:
would it make more sense to use `ids.ID` rather than `[dataLen]byte`?

Reply:
i feel like we would lose the notion of "arbitrary byte/data" if we use `ids.ID`.

Reply:
all an `ids.ID` is - is a [32]byte. Which is the same thing though